extra_data/write_cxi.py

```diff
@@ -21,15 +21,21 @@ class VirtualCXIWriter:
     Parameters
     ----------
 
-    detdata: extra_data.components.MPxDetectorBase
+    detdata: extra_data.components.XtdfDetectorBase
       The detector data interface for the data to gather in this file.
     """
     def __init__(self, detdata):
         self.detdata = detdata
 
         self.modulenos = sorted(detdata.modno_to_source)
 
-        frame_counts = detdata.frame_counts
+        # The following line could be tricky, but there is no pulseId
+        # in Jungfrau data.
+        self.pulse_id_label = 'image.pulseId'
+        self.group_label, self.image_label = detdata._main_data_key.split('.')
+        self.cell_id_label = 'cellId'
```

Comment:
It would probably be clearer to have a base class which doesn't set these labels, and then two subclasses (one for JUNGFRAU, one for xtdf detectors), which know their respective names, rather than the base class being used directly for xtdf detectors.

There are probably also some complexities which are only needed for xtdf detectors and can be moved to that subclass.

Reply:
I've tried to introduce such separation into Base, Xtdf and JUNGFRAU classes in https://github.com/European-XFEL/EXtra-data/commit/cc73dadb190a2c3e831a1fb08259f2752871ee5d - please let me know your opinion.